src/Symfony/Bundle/FrameworkBundle/Test/KernelTestCase.php

```diff
@@ -165,10 +213,35 @@ protected static function createKernel(array $options = array())
             static::$class = static::getKernelClass();
         }
 
-        return new static::$class(
+        if (isset($_SERVER['KERNEL_ENV'])) {
+            $options['environment'] = $_SERVER['KERNEL_ENV'];
```

Comment:
Passing options explicitly must still be respected. Otherwise, it is just a WTF.